vega_sim/scenario/fuzzed_markets/agents.py

```diff
@@ -0,0 +1,347 @@
+from vega_sim.environment.agent import StateAgentWithWallet
+from typing import Optional
+from vega_sim.null_service import VegaServiceNull
+from numpy.random import RandomState
+from vega_sim.proto.vega import markets as markets_protos
+
+
+class FuzzingAgent(StateAgentWithWallet):
+    NAME_BASE = "fuzzing_agent"
+
+    def __init__(
+        self,
+        key_name: str,
+        market_name: str,
+        asset_name: str,
+        tag: Optional[str] = None,
+        wallet_name: Optional[str] = None,
+        state_update_freq: Optional[int] = None,
+        initial_asset_mint: float = 1e9,
+        random_state: Optional[RandomState] = None,
+    ):
+        super().__init__(
+            key_name=key_name,
+            tag=tag,
+            wallet_name=wallet_name,
+            state_update_freq=state_update_freq,
+        )
+
+        self.market_name = market_name
+        self.asset_name = asset_name
+        self.initial_asset_mint = initial_asset_mint
+        self.random_state = random_state if random_state is not None else RandomState()
+
+    def initialise(
+        self, vega: VegaServiceNull, create_key: bool = True, mint_key: bool = True
+    ):
+        super().initialise(vega, create_key)
+
+        self.market_id = self.vega.find_market_id(name=self.market_name)
+
+        # Get asset id
+        self.asset_id = self.vega.find_asset_id(symbol=self.asset_name)
+        if mint_key:
+            # Top up asset
+            self.vega.mint(
+                key_name=self.key_name,
+                asset=self.asset_id,
+                amount=self.initial_asset_mint,
+                wallet_name=self.wallet_name,
+            )
+
+        self.vega.wait_fn(5)
+
+    def step(self, vega_state):
+        self.live_orders = self.vega.orders_for_party_from_feed(
+            key_name=self.key_name,
+            wallet_name=self.wallet_name,
+            market_id=self.market_id,
+            live_only=True,
+        )
+
+        self.curr_price = vega_state.market_state[self.market_id].midprice
+
+        submissions = [self.create_fuzzed_submission() for _ in range(20)]
+        amendments = [self.create_fuzzed_amendment() for _ in range(10)]
+        cancellations = [self.create_fuzzed_cancellation() for _ in range(1)]
+
+        self.vega.submit_instructions(
+            key_name=self.key_name,
+            wallet_name=self.wallet_name,
+            submissions=submissions,
+            amendments=amendments,
+            cancellations=cancellations,
+        )
+
+    def create_fuzzed_cancellation(self):
+        if self.live_orders != {}:
+            order_key = self.random_state.choice(list(self.live_orders.keys()))
+            order_id = self.live_orders[order_key].id
+        else:
+            order_id = None
```

Comment:
Could all be:
`order_id = self.random_state.choice(list(self.live_orders.values())).id if self.live_orders else None`

Reply:
I tried something similar initially but ran into issues with numpy not liking lists of objects. Although maybe I am missing something obvious?
```
(vega-sim-py3.10) Charlies-MacBook-Pro:vega-market-sim caed$ /Users/caed/vega-market-sim/.venv/bin/python /Users/caed/vega-market-sim/vega_sim/scenario/fuzzed_markets/scenario.py
INFO:vega_sim.null_service:Vega Running. Console launched at http://localhost:50337
INFO:vega_sim.null_service:Running NullChain from vegahome of /var/folders/mq/qntzn9jd0rngp7d0cj8nz9xh0000gn/T/vega-sim-o152z5ze
INFO:vega_sim.null_service:Launching GraphQL node at port 50331
INFO:vega_sim.null_service:Launching Console at port 50337
INFO:vega_sim.null_service:Launching Datanode REST at port 50330
INFO:vega_sim.null_service:Launching Datanode GRPC at port 50329
INFO:vega_sim.null_service:Launching Core GRPC at port 50335
INFO:vega_sim.environment.environment:Running wallet at: http://localhost:50328
INFO:vega_sim.environment.environment:Running graphql at: http://localhost:50331
Traceback (most recent call last):
  File "mtrand.pyx", line 924, in numpy.random.mtrand.RandomState.choice
TypeError: 'dict_values' object cannot be interpreted as an integer

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/caed/vega-market-sim/vega_sim/scenario/fuzzed_markets/scenario.py", line 246, in <module>
    scenario.run_iteration(
  File "/Users/caed/vega-market-sim/vega_sim/scenario/scenario.py", line 79, in run_iteration
    outputs = self.env.run(
  File "/Users/caed/vega-market-sim/vega_sim/environment/environment.py", line 150, in run
    return self._run(self._vega, pause_at_completion=pause_at_completion)
  File "/Users/caed/vega-market-sim/vega_sim/environment/environment.py", line 200, in _run
    self.step(vega)
  File "/Users/caed/vega-market-sim/vega_sim/environment/environment.py", line 373, in step
    agent.step(state)
  File "/Users/caed/vega-market-sim/vega_sim/scenario/fuzzed_markets/agents.py", line 67, in step
    cancellations = [self.create_fuzzed_cancellation() for _ in range(1)]
  File "/Users/caed/vega-market-sim/vega_sim/scenario/fuzzed_markets/agents.py", line 67, in <listcomp>
    cancellations = [self.create_fuzzed_cancellation() for _ in range(1)]
  File "/Users/caed/vega-market-sim/vega_sim/scenario/fuzzed_markets/agents.py", line 79, in create_fuzzed_cancellation
    self.random_state.choice(array(self.live_orders.values())).id
  File "mtrand.pyx", line 926, in numpy.random.mtrand.RandomState.choice
ValueError: a must be 1-dimensional or an integer
(vega-sim-py3.10) Charlies-MacBook-Pro:vega-market-sim caed$ 
```

Reply:
Oh fun, yeah I guess let's just leave as-is then!